packages/data/addon/services/navi-dimension.ts

```diff
@@ -45,6 +160,13 @@ export default class NaviDimensionService extends Service {
    * @param options - method options
    */
   @task *all(dimension: DimensionColumn, options: ServiceOptions = {}): TaskGenerator<NaviDimensionResponse> {
+    // check cache
+    const cacheId = this._getCacheId(dimension);
+    const cacheResponse = this._checkCache(cacheId);
```

Comment:
Can we call `getCacheId` inside `_checkCache` so `all` doesn't need to know the details of the cache?

Reply:
Yes, but then getCacheId will be called twice for any instance where something isn't in the cache (once to check the cache at the beginning, and then again when adding the new records to the cache). If that's nbd, though, I can change it